spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/HibernateProperties.java

```diff
@@ -129,6 +130,9 @@ private void applyScanner(Map<String, Object> result) {
 	}
 
 	private String determineDdlAuto(Map<String, String> existing, Supplier<String> defaultDdlAuto) {
+        if (existing.get(AvailableSettings.HBM2DDL_DATABASE_ACTION) != null) {
```

Comment:
I made the db-action property take precedence over the hbm property to be consistent w/ [Hibernate](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/tool/schema/spi/SchemaManagementToolCoordinator.java#L488).